### ISSUE TITLE
rdflib.util.guess_format does not detect jsonld

### ISSUE THREAD

**Issue by tcmitchell** [RDFLib/rdflib#1412]
Prior to RDFLib 6, when rdflib_jsonld was a separate package and had to be imported separately, `rdflib.util.guess_format` would detect 'jsonld' as the suffix for JSON-LD format.

[RDFLib 6 incorporates rdflib_jsonld](https://github.com/RDFLib/rdflib-jsonld#archived) but `rdflib.util.guess_format` no longer detects 'jsonld' as the suffix for JSON-LD files.

```
$ python3
Python 3.8.5 (v3.8.5:580fbb018f, Jul 20 2020, 12:11:27) 
[Clang 6.0 (clang-600.0.57)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import rdflib
RDFLib Version: 6.0.0
>>> rdflib.util.SUFFIX_FORMAT_MAP
{'xml': 'xml', 'rdf': 'xml', 'owl': 'xml', 'n3': 'n3', 'ttl': 'turtle', 'nt': 'nt', 'trix': 'trix', 'xhtml': 'rdfa', 'html': 'rdfa', 'svg': 'rdfa', 'nq': 'nquads', 'trig': 'trig'}
>>> 'jsonld' in rdflib.util.SUFFIX_FORMAT_MAP
False
```

It is now necessary to import the json-ld parser separately, just as it was in the old days.

```
>>> import rdflib.plugins.parsers.jsonld
>>> rdflib.util.SUFFIX_FORMAT_MAP
{'xml': 'xml', 'rdf': 'xml', 'owl': 'xml', 'n3': 'n3', 'ttl': 'turtle', 'nt': 'nt', 'trix': 'trix', 'xhtml': 'rdfa', 'html': 'rdfa', 'svg': 'rdfa', 'nq': 'nquads', 'trig': 'trig', 'jsonld': 'application/ld+json'}
>>> 'jsonld' in rdflib.util.SUFFIX_FORMAT_MAP
True
```

This feels wrong. Why does the JSON-LD parser have to be imported separately to make `rdflib.util.guess_format` detect the JSON-LD format?

Can JSON-LD be an automatically detected format out of the box now that the JSON-LD parser is included in RDFLib?

**Comment by ajnelson-nist** [comment]
Hello @tcmitchell ,

I filed a PR a few days ago that updated this functionality.
https://github.com/RDFLib/rdflib/pull/1403

It seems this ticket can be resolved on a release incorporating that PR.

It's interesting that this used to work transparently for you.  I had to write a helper function, which I'll be happy to remove as "Tech-transferred" when that release is made.

**Comment by nicholascar** [comment]
I released rdflib 6.0.1 last night that contains PR #1403 so I think this issue is solved. I'll close the ticket now but please reopen/raise another if you notice problems.
